docs/3x/quantization.md

```diff
@@ -48,50 +65,325 @@ If UINT8 is specified, $Scale = max(abs(X_{f_{max}}), abs(X_{f_{min}})) / 255$ a
+
+The image on the left presents a normal linear forward  with 1x2 input $x$ and 2x2 weight $w$. The results $y$ could be easily obtained by simple mathematics. In the middle image, we apply per-tensor quantization for activations and per-channel quantization for weights; the results after quantization that are denoted by $y_1$ and $y_2$, could be easily dequantized to the float results $y_{fp1}$ and $y_{fp2}$ by per channel scale $1.0/s_1s_x$ and $1.0/s_2s_x$. However, after applying per-channel quantization for activation (right image), we could not dequantize the  $y_1$ and  $y_2$ to float results.
+
+<div align="center">
+    <img src="../imgs/sq_pc.png"/>
+</div>
+
+
+----------------------------------------------
+
+In the previous subsection, we have explained why per-channel quantization could not be applied for activation, even though it could lead to lower quantization loss. However, the quantization error loss of activation plays an important role in the accuracy loss of model quantization[1][6][7]. 
+
+
+
+To reduce the quantization loss of activations, lots of methods have been proposed. In the following, we briefly introduce SPIQ[6], Outlier Suppression[7] and Smoothquant[1]. All these three methods share a similar idea to migrate the difficulty from activation quantization to weight quantization but differ in how much difficulty to be transferred.
+
+
+So **the first question is how to migrate the difficulty from activation to weights?** The solution is straightforward, that is to convert the network to an output equivalent network that is presented in the image below and apply quantization to this equivalent network. The intuition is that each channel of activation could be scaled to make it more quantization-friendly, similar to a fake per-channel activation quantization.
+
+<div align="center">
+    <img src="../imgs/sq_convert.png"/>
+</div>
+
+
+Please note that this conversion will make the quantization of weights more difficult, because the scales attached to weights shown above are per-input-channel, while quantization of weights is per-output-channel or per-tensor.
+
+So **the second question is how much difficulty to be migrated**, that is how to choose the **conversion per-channel scale** $s_{x1}$ and $s_{x2}$ from the above image. Different works adopt different ways.
+
+*SPIQ* just adopts the quantization scale of activations as the conversion per-channel scale.
+
+*Outlier suppression* adopts the scale of the preceding layernorm as the conversion per-channel scale.
+
+*Smoothquant* introduces a hyperparameter $\alpha$ as a smooth factor to calculate the conversion per-channel scale and balance the quantization difficulty of activation and weight.
+
+$$
+s_j = max(|X_j|)^\alpha/max(|W_j|)^{1-\alpha} \tag{4}
+$$
+
+j is the index of the input channels.
+
+
+
+<div align="center">
+    <img src="../imgs/smoothquant.png" height="250"/>
```

Comment:
same as above, link is broken

Reply:
Thanks for checking this.
It's my fault, I shouldn't have moved the folder without checking, I will fix it soon.